### PR TITLE
SSR rendering fix for amp-list

### PR DESCRIPTION
**Pull request by alabiaga**
only pass in child nodes of the ssr template response when updating bindings and rendering. When passing it in its entirety, the list container class attribute is stripped by the sanitizer/purifier in the rendering phase.

e.g. Pass in list items

    <div role="list-item"></div>
    <div role="list-item"></div>
    <div role="list-item"></div>

vs list container with list items

    <div role="list" ..>
      <div role="list-item"></div>
      <div role="list-item"></div>
      <div role="list-item"></div>
    </div>

- update ssr related amp documents in examples.
- update amp-list ssr response to an actual expected amp-list container and list items format.

b/123944762

### CODE REVIEW

**Comment by alabiaga** [comment]
cc\ @GoTcWang

File: extensions/amp-list/0.1/amp-list.js [hunk 1]
```diff
@@ -567,7 +567,12 @@ export class AmpList extends AMP.BaseElement {
     const isSSR = this.ssrTemplateHelper_.isSupported();
     let renderPromise = this.ssrTemplateHelper_.renderTemplate(
         this.element, current.data)
-        .then(result => this.updateBindings_(result, current.append))
+        // For SSR, the result will be the container node that contains the
+        // list items. Just pass in the list items when updating the bindings
+        // and rendering else the sanitizer will strip out the class attribute
+        // from the container.
+        .then(result => this.updateBindings_(
+            isSSR ? toArray(result.childNodes) : result, current.append))
```
Comment (choumx) [inline extensions/amp-list/0.1/amp-list.js]: What value of `class` attribute is being stripped? DOMPurify itself doesn't strip the `class` attribute, but maybe we are sanitizing the specific value.

Comment (alabiaga) [reply extensions/amp-list/0.1/amp-list.js]: Right, it is actually behaving as it should, stripping the following amp reserved css.

    <div role="list" class="**i-amphtml-fill-content i-amphtml-replaced-content**">
       <div class="product" role="listitem">Apple</div>
       ...
    </div>

**Comment by alabiaga** [comment]
> Surprised that no unit tests need to change for this. Let's add one if it's missing?

Right, the unit tests only verified that the right calls were being made and didn't verify the output of the rendered html. I'll add them. Thanks Will.

File: extensions/amp-list/0.1/amp-list.js [hunk 2]
```diff
@@ -619,15 +624,12 @@ export class AmpList extends AMP.BaseElement {
    * Scans for, evaluates and applies any bindings in the given elements.
    * Ensures that rendered content is up-to-date with the latest bindable state.
    * Can be skipped by setting binding="no" or binding="refresh" attribute.
-   * @param {!Array<!Element>|!Element} elementOrElements
+   * @param {!Array<!Element>} elements
```
Comment (cathyxz) [inline extensions/amp-list/0.1/amp-list.js]: Good change.

**Pull request merged**